fastqpreprocessing/src/htslib_tagsort.cpp

```diff
@@ -0,0 +1,428 @@
+/**
+ *  @file   htslib_tagsort.cpp
+ *  @brief  functions for file processing
+ *  @author Kishori Konwar
+ *  @date   2021-08-11
+ ***********************************************/
+
+#include "htslib_tagsort.h"
+
+unsigned int num_thread_deallocations = 0;
+extern sem_t semaphore;
+extern std::mutex mtx;
+extern std::set<unsigned int> busy_buffer, idle_buffer, threads_to_join;
+
+char*  error_sem_wait = "sem_wait: semaphore"; 
+char*  error_sem_post = "sem_post: semaphore"; 
+
+#define SEM_PRINT_VAL(X,Y)                               \
```

Comment:
Have you checked this code for race conditions?

Reply:
still curious

Reply:
It does not apply here, using mutex avoids race condition.